### PR TITLE
nit: Cleanup auth index tests

### PR DESCRIPTION
**Pull request by schew2381**
1. Import setting `DISABLE_SSO_CHECK_FOR_LOCAL_DEV` directly from superuser b/c it's simpler and makes it easier to test.
2. Fix tests that fake tested this setting by overriding self-hosted call instead of the actual setting.
3. Also cleanup imports and remove double overrides of self-hosted
4. Add some extra error response checking to see what the actual error message is

This PR looks big but due to getting rid of all the `with` nesting, no real code change

### CODE REVIEW

File: tests/sentry/api/endpoints/test_auth_index.py [hunk 1]
```diff
@@ -164,374 +168,340 @@ def test_invalid_password_u2f(self, validate_response, is_available, mock_metric
 class AuthVerifyEndpointSuperuserTest(AuthProviderTestCase, APITestCase):
     path = "/api/0/auth/"
 
+    @override_settings(SENTRY_SELF_HOSTED=False)
     @with_feature("organizations:u2f-superuser-form")
     @mock.patch("sentry.auth.authenticators.U2fInterface.is_available", return_value=True)
     @mock.patch("sentry.auth.authenticators.U2fInterface.validate_response", return_value=True)
     def test_superuser_sso_user_no_password_saas_product(self, validate_response, is_available):
-        from sentry.auth.superuser import COOKIE_NAME, Superuser
```
Comment (cathteng) [inline tests/sentry/api/endpoints/test_auth_index.py:171]: weird that we were repeatedly importing this

File: tests/sentry/api/endpoints/test_auth_index.py [hunk 2]
```diff
@@ -164,374 +168,340 @@ def test_invalid_password_u2f(self, validate_response, is_available, mock_metric
...
+                self.path,
+                data={
+                    "isSuperuserModal": True,
+                    "superuserReason": "for testing",
+                },
+            )
+            assert response.status_code == 403
+            assert response.data["detail"]["code"] == "ignore"
 
+    @override_settings(SENTRY_SELF_HOSTED=False, VALIDATE_SUPERUSER_ACCESS_CATEGORY_AND_REASON=True)
```
Comment (cathteng) [inline tests/sentry/api/endpoints/test_auth_index.py:338]: `VALIDATE_SUPERUSER_ACCESS_CATEGORY_AND_REASON` is already `True` in sentry? https://github.com/getsentry/sentry/blob/a9895178b56b9cf20e4b4d18d09ff0b005ec230d/src/sentry/conf/server.py#L3795

**Review by cathteng — APPROVED** [review]
lgtm, just that one nit

File: tests/sentry/api/endpoints/test_auth_index.py (hunk 2 above)
Comment (schew2381) [reply tests/sentry/api/endpoints/test_auth_index.py]: ah youre right, got rid of that too

**Pull request merged**